homeassistant/components/sensor/gitlab_ci.py

```diff
@@ -0,0 +1,182 @@
+    ATTR_ATTRIBUTION, CONF_SCAN_INTERVAL, CONF_TOKEN, STATE_UNKNOWN)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+CONF_GITLAB_ID = 'gitlab_id'
+CONF_ATTRIBUTION = "Information provided by https://gitlab.com/"
+
+ICON_HAPPY = 'mdi:emoticon-happy'
+ICON_SAD = 'mdi:emoticon-happy'
+ICON_OTHER = 'mdi:git'
+
+ATTR_BUILD_ID = 'build id'
+ATTR_BUILD_STATUS = 'build_status'
+ATTR_BUILD_STARTED = 'build_started'
+ATTR_BUILD_FINISHED = 'build_finished'
+ATTR_BUILD_DURATION = 'build_duration'
+ATTR_BUILD_COMMIT_ID = 'commit id'
+ATTR_BUILD_COMMIT_DATE = 'commit date'
+ATTR_BUILD_BRANCH = 'master'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Required(CONF_GITLAB_ID): cv.string,
+    vol.Optional(CONF_SCAN_INTERVAL,
+                 default=timedelta(seconds=300)): cv.time_period,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Sensor platform setup."""
+    _logger = logging.getLogger(__name__)
+
+    _priv_token = config.get(CONF_TOKEN)
+    _gitlab_id = config.get(CONF_GITLAB_ID)
+    _interval = config.get(CONF_SCAN_INTERVAL)
+
+    if _priv_token is None:
+        _logger.error('No private access token specified')
+        return False
+    if _gitlab_id is None:
+        _logger.error('No GitLab ID specified')
+        return False
+
+    _gitlab_data = GitLabData(
+        priv_token=_priv_token,
+        gitlab_id=_gitlab_id,
+        interval=_interval
+    )
+
+    add_devices([GitLabSensor(_gitlab_id, _priv_token, _gitlab_data)])
+
+
+class GitLabSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, gitlab_id, priv_token, gitlab_data):
+        """Initialize the sensor."""
+        self._state = None
+        self._gitlab_data = gitlab_data
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return 'GitLab CI Status'
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+            ATTR_BUILD_STATUS: self._status,
+            ATTR_BUILD_STARTED: self._started_at,
+            ATTR_BUILD_FINISHED: self._finished_at,
+            ATTR_BUILD_DURATION: self._duration,
+            ATTR_BUILD_COMMIT_ID: self._commit_id,
+            ATTR_BUILD_COMMIT_DATE: self._commit_date,
+            ATTR_BUILD_ID: self._build_id,
+            ATTR_BUILD_BRANCH: self._branch
+        }
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        if self._status == 'success':
+            return ICON_HAPPY
+        if self._status == 'failed':
+            return ICON_SAD
+        return ICON_OTHER
+
+    def update(self):
```

Comment:
if there has been an error, you need to return `True` from the `available` property.

Reply:
Added `available` property